postman/src/application/postman/persistence/subscribers/MessageStatusSubscriber.ts

```diff
@@ -0,0 +1,165 @@
+      if (previousStatus !== newStatus) {
+        await this.updateMessageMetricsOnUpdate(event);
+      }
+    }
+  }
+
+  async afterRemove(event: RemoveEvent<MessageEntity>): Promise<void> {
+    if (event.entity) {
+      await this.updateMessageMetricsOnRemove(event);
+    }
+  }
+
+  async afterTransactionCommit(event: TransactionCommitEvent): Promise<void> {
+    const updatedEntity = event.queryRunner?.data?.updatedEntity;
+    if (updatedEntity) {
+      const [previousGauge, gauge] = await Promise.all([
+        this.metricsService.getGaugeValue("postman_messages_current", {
+          status: updatedEntity.previousStatus,
+          direction: updatedEntity.direction,
+        }),
+        this.metricsService.getGaugeValue("postman_messages_current", {
+          status: updatedEntity.newStatus,
+          direction: updatedEntity.direction,
+        }),
+      ]);
+
+      if (previousGauge && previousGauge > 0) {
+        this.metricsService.decrementGauge("postman_messages_current", 1, {
+          status: updatedEntity.previousStatus,
+          direction: updatedEntity.direction,
+        });
+      }
+
+      if (gauge || gauge === 0) {
+        this.metricsService.incrementGauge("postman_messages_current", 1, {
+          status: updatedEntity.newStatus,
+          direction: updatedEntity.direction,
+        });
+      }
+    }
+  }
+
+  private async updateMessageMetricsOnInsert(event: InsertEvent<MessageEntity>): Promise<void> {
+    try {
+      const messageStatus = event.entity.status;
+      const messageDirection = event.entity.direction;
+
+      const gauge = await this.metricsService.getGaugeValue("postman_messages_current", {
+        status: messageStatus,
+        direction: messageDirection,
+      });
+
+      if (!gauge && gauge !== 0) {
+        return;
+      }
+
+      this.metricsService.incrementGauge("postman_messages_current", 1, {
+        status: messageStatus,
+        direction: messageDirection,
+      });
+    } catch (error) {
+      this.logger.error("Failed to update metrics:", error);
+    }
+  }
+
+  private async updateMessageMetricsOnUpdate(event: UpdateEvent<MessageEntity>): Promise<void> {
+    try {
+      if (!event.entity) {
+        return;
+      }
+      const messageStatus = event.entity.status;
+      const previousStatus = event.databaseEntity.status;
+      const messageDirection = event.databaseEntity.direction;
+
+      const [previousGauge, gauge] = await Promise.all([
+        this.metricsService.getGaugeValue("postman_messages_current", {
+          status: previousStatus,
+          direction: messageDirection,
+        }),
+        this.metricsService.getGaugeValue("postman_messages_current", {
+          status: messageStatus,
+          direction: messageDirection,
+        }),
+      ]);
+
+      if (!previousGauge && previousGauge !== 0) {
+        return;
+      }
+
+      if (!gauge && gauge !== 0) {
+        return;
+      }
```

Comment:
```suggestion
      if (previousGauge === undefined || gauge === undefined) {
        return;
      }
```

Reply:
I also think these two condition checks are confusing, just a check for `undefined` is more intuitive